### DISCUSSION TITLE
Comment Preservation

### DISCUSSION THREAD

**Discussion by lokka30** [SpongePowered/Configurate#280] [Support] [answered]
Hi there!

Does Configurate support any form of comment preservation? i.e., when any changes are written to a file, the comments in the file are preserved in the same location?

Hope I didn't ask a question that was already answered here, I tried searching but nothing came up.

**Comment by zml2008** [comment, marked as answer]
Yes -- the HOCON and XML formats currently roundtrip comments fairly reliably, and there is work in progress to support the same for the YAML loader in #175. These loaders will all give you `CommentedConfigurationNode`s which provide comment information for tinkering with.

When working with the object mapper, the `@Comment` annotation will allow you to provide initial comments for mapped fields.

**Reply by lokka30** [reply]
Thank you so much.
